File: Curso 1/T7/Ejercicios Clase/Ejercicio 2 T6/src/model/Autonomo.java
package model;

public class Autonomo extends Trabajador{

    private double sueldo;
    private boolean contratado;


    public Autonomo(){

    }

    public Autonomo(String nombre, String apellido, String dni, double sueldo, boolean contratado) {
        super(nombre, apellido, dni);
        this.sueldo = sueldo;
        this.contratado = contratado;
    }


    @Override
    public void mostrarDatos() {
        super.mostrarDatos();
        System.out.println(" Sueldo: " + sueldo + " Contratado: " + contratado);
    }

    public double getSueldo() {
        return sueldo;
    }

    public void setSueldo(double sueldo) {
        this.sueldo = sueldo;
    }

    public boolean isContratado() {
        return contratado;
    }

    public void setContratado(boolean contratado) {
        this.contratado = contratado;
    }
}

File: Curso 2/PMPD/Tema1/TrabajadoresHerencia/src/model/Autonomo.kt
package model

class Autonomo(
    nombre: String,
    apellido: String,
    dni: String,
    salario: Double,
    seguro: Boolean,
    var cuotasSS: Int,
) : Trabajador(nombre, apellido, dni, salario, seguro = seguro) {

    override fun calcularSalarioNeto(): Double {
        val salarioNeto: Double = salario - (cuotasSS * 12)
        return salarioNeto
    }

    fun pedirDescuento() {
        if (seguro == true){
            cuotasSS  *= 0.5.toInt()
            println("Te hemos descontado las cuotas a la mitad")
        } else {
            println("No tienes seguro por lo que no te hemos condedido ningun descuento")
        }
    }

    override fun mostrarDatos() {
        super.mostrarDatos()
        println("CuotasSS: $cuotasSS")
    }

}